server/lib/src/plugins/ecdhkeygen.rs

```diff
@@ -0,0 +1,142 @@
+use openssl::ec::{EcGroup, EcKey};
+use openssl::nid::Nid;
+use sketching::admin_error;
+
+use super::Plugin;
+use crate::event::CreateEvent;
+use crate::prelude::{Entry, EntryInvalidNew, OperationError};
+use crate::server::QueryServerWriteTransaction;
+use crate::value::PartialValue;
+use sketching::tagged_event;
+use sketching::EventTag;
+
+lazy_static! {
+    // it contains all the partialvalues used to match against an Entry's class,
+    // we just need a partialvalue to match in order to target the entry
+    static ref CLASSES_TO_UPDATE: [PartialValue; 1] = [PartialValue::new_iutf8("account")];
+    static ref DEFAULT_KEY_GROUP: EcGroup = {
+        let nid = Nid::X9_62_PRIME256V1; // NIST P-256 curve
+        EcGroup::from_curve_name(nid).unwrap()
+    };
+}
+pub struct EcdhKeyGen {}
+
+impl EcdhKeyGen {
+    fn is_entry_to_update<VALUE, STATE>(entry: &mut Entry<VALUE, STATE>) -> bool {
+        CLASSES_TO_UPDATE
+            .iter()
+            .any(|pv| entry.attribute_equality("class", pv))
+    }
+    fn handle_user_account_creation(cands: &mut [EntryInvalidNew]) -> Result<(), OperationError> {
+        for cand in cands.iter_mut() {
+            if Self::is_entry_to_update(cand) {
+                let new_key = EcKey::generate(&DEFAULT_KEY_GROUP)
+                    .map_err(|_| OperationError::CryptographyError)?;
+                let new_key_der = new_key
+                    .private_key_to_der()
+                    .map_err(|_| OperationError::CryptographyError)?;
+                cand.add_ava_if_not_exist(
+                    "id_verification_eckey",
+                    crate::value::Value::ECDHKey(new_key_der),
+                )
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl Plugin for EcdhKeyGen {
+    fn id() -> &'static str {
+        "plugin_ecdhkey_gen"
+    }
+    fn pre_create_transform(
+        _qs: &mut QueryServerWriteTransaction,
+        cand: &mut Vec<EntryInvalidNew>,
+        _ce: &CreateEvent,
+    ) -> Result<(), OperationError> {
+        Self::handle_user_account_creation(cand)
+    }
+
+    fn pre_modify(
```

Comment:
This needs to handle mods too. Because else how do we generate keys for users that already exist?